### PR TITLE
Fixes #22472 - added basic telemetry

### PR DESCRIPTION
**Pull request by lzap**
An example how plugin can leverage the new telemetry API in core.

This needs https://github.com/theforeman/foreman/pull/5096

### CODE REVIEW

**Comment by lzap** [comment]
Rebased, @rahulbajaj0509 this should be fun.

**Comment by rabajaj0509** [comment]
@lzap can you resolve the conflicts and rebase this pr?

**Comment by lzap** [comment]
Done!

**Comment by lzap** [comment]
Pushed, anyone? @ares

**Comment by lzap** [comment]
Rebased this should pass now @rahulbajaj0509

**Comment by lzap** [comment]
Error when running tests:

    /home/lzap/work/foreman_hooks/lib/foreman_hooks/as_dependencies_hook.rb:4:in `load_missing_constant': uninitialized constant Foreman::Plugin::DEFAULT_BUCKETS (NameError)

**Comment by lzap** [comment]
Fixed, tests will fail @rahulbajaj0509 until a trivial fix lands in core: https://github.com/theforeman/foreman/pull/6233

**Comment by ares** [comment]
[test foreman_discovery] the patch has landed

**Comment by lzap** [comment]
Rebased, mind to look? Thanks.

**Comment by rabajaj0509** [comment]
[test foreman_discovery]

**Comment by lzap** [comment]
I just rebased this old lady.

File: app/models/host/discovered.rb [hunk 1]
```diff
@@ -126,7 +132,12 @@ def populate_fields_from_facts(parser, type, source_proxy)
   def populate_discovery_fields_from_facts(facts)
     ForemanDiscovery::ImportHookService.new(host: self, facts: facts).after_populate
   ensure
-    self.save!
+    begin
+      self.save!
+    rescue => _
```
Comment (rabajaj0509) [inline app/models/host/discovered.rb:143]: I am guessing this `_` was used while testing in IRB?

File: app/models/host/discovered.rb [hunk 2]
```diff
@@ -126,7 +132,12 @@ def populate_fields_from_facts(parser, type, source_proxy)
...
+      telemetry_increment_counter(:failed_discovered_hosts)
+      raise
```
Comment (rabajaj0509) [inline app/models/host/discovered.rb:145]: Would you want to raise an error or default to `StandardError`?

**Review by rabajaj0509 — COMMENTED** [review]
Code wise looks good! Need to test the functionality though.

File: app/models/host/discovered.rb (hunk 2 above)
Comment (lzap) [reply app/models/host/discovered.rb:145]: Calling raise without any arguments will raise the last error. In our case, we are re-raising error. This is a standard idiom in many programming languages, I have this from Java and I believe Ruby does the same.

File: app/models/host/discovered.rb (hunk 1 above)
Comment (lzap) [reply app/models/host/discovered.rb:143]: Well this is to trick Rubocop that this variable will be unused. It is a common idiom in other languages, I think it also makes Rubocop happy.

**Comment by stejskalleos** [comment]
There are some merge conflicts, @lzap could you rebase?

**Comment by lzap** [comment]
Done.

**Comment by lzap** [comment]
Closing, feel free to take it from here if you think this patch is useful. Cheers!

**Pull request closed without merging**